File: Code Workspace/C - Codes/binarysearch.c
#include <stdio.h>
int binarysearch(int arr[],int mid,int lw,int up,int k);
int binarysearch(int arr[],int mid,int lw,int up,int k)
{
    mid = (lw + up)/2;
    if(arr[mid]==k)
    {
        return mid;
    }
    else if(k>arr[mid])
    {
        return binarysearch(arr,mid,mid+1,up,k);
    }
    else if(k<arr[mid])
    {
        return binarysearch(arr,mid,lw,mid-1,k);
    }
    else
    return -1;

}
int main()
    {

int arr[10];
for(int i = 0;i< 10;i++)
{
   scanf("%d",&arr[i]);
}
int l=0;
int u = 9;
int mid = (l + u)/2;

int key;
scanf("%d",&key);
int a = binarysearch(arr,mid,l,u,key);
printf("%d",a);
    }

File: Code Workspace/Java Kunal Kushwaha/Searching and Sorting/Binarysearch.java
import java.util.Scanner;

public class Binarysearch {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the size of the array");
        int size = sc.nextInt();
        int[]arr = new int[size];
        for(int i = 0 ; i<size ; i++)
        {
            arr[i] = sc.nextInt();
        }
        System.out.println("Enter the element to search: ");
        int s = sc.nextInt();
        int ans = bs(arr,s);
        System.out.println("The element is present at the " + ans + " position");
        sc.close();
    }
    static int bs(int[]arr,int target)
    {
         int low = 0;
         int high = arr.length -1;
         while(low<=high)
         {
            int mid = low + (high-low)/2; //this is a better way to find mid cause it wont exceed the integer range in case low and high are bigger numbers
            if(target<arr[mid])
            {
                 high = mid-1;
            }
            else if(target>arr[mid])
            {
                low = mid+1;
            }
            else
            {
                return mid;
            }
         }
         return -1;
    }
}
